kong/core/reports.lua

```diff
@@ -52,6 +52,22 @@ end
 
 -- UDP logger
 
+local function get_report_value(v)
+  if type(v) == "function" then
+    v = v()
+  end
+
+  if type(v) == "table" then
+    local json, err = cjson.encode(v)
+    if err then
+      log(ERR, "could not JSON encode given table entity: ", err)
+    end
+
+    v = json
+  end
+
+  return v
```

Comment:
minor: we could do `return v ~= nil and tostring(v) or nil` here, and provide a safer API (both appends to the buffer would not have to do the `tostring()` themselves anymore, nor would any future code change, but not a blocker.